lib/Map/UserDrawing.js

```diff
@@ -0,0 +1,400 @@
+                    that.onPointClicked(that.pointEntities);
+                }
+            }
+        });
+    });
+};
+
+/**
+ * Create the HTML message in the dialog box.
+ * Example:
+ *
+ *     Measuring Tool
+ *     373.45 km
+ *     Click to add another point
+ *
+ * @private
+ */
+UserDrawing.prototype._getDialogMessage = function() {
+
+    var message = "<strong>" + this.messageHeader + "</strong></br>";
+
+    var innerMessage = "";
+    if (typeof this.onMakeDialogMessage === "function") {
+        innerMessage = this.onMakeDialogMessage();
+    }
+    if (innerMessage !== "") {
+        message += innerMessage + "</br>";
+    }
+
+    var word = "a";
+    if (this.pointEntities.entities.values.length > 0) {
+        word = "another";
+    }
+    message += "<i>Click to add " + word + " point</i>";
+    // htmlToReactParser will fail if html doesn't have only one root element.
+    return "<div>" + message + "</div>";
+};
+
+/**
+ * Figure out the text for the dialog button.
+ * @private
+ */
+UserDrawing.prototype._getButtonText = function() {
+    var buttonText = "Cancel";
+    if (this.pointEntities.entities.values.length >= 2) {
+        buttonText = "Done";
+    }
+    return buttonText;
+};
+
+/**
+ * User has finished or cancelled; restore initial state.
+ * @private
+ */
+UserDrawing.prototype._cleanUp = function() {
+    this.terria.dataSources.remove(this.pointEntities);
+    this.pointEntities = new CustomDataSource('Points');
+    this.terria.dataSources.remove(this.otherEntities);
+    this.otherEntities = new CustomDataSource('Lines and polygons');
+
+    this.inDrawMode = false;
+    this.closeLoop = false;
+
+    // Return cursor to original state
+    document.getElementById("cesiumContainer").setAttribute("style", "cursor: auto");
+
+    // Allow client to clean up too
+    if (typeof this.onCleanUp === "function") {
+        this.onCleanUp();
+    }
+};
+
+/**
+ * Called after a point has been added, this updates the MapInteractionModeStack with a listener for another point.
+ * @private
+ */
+UserDrawing.prototype._mapInteractionModeUpdate = function() {
+    this.terria.mapInteractionModeStack.pop();
+    var that = this;
+    const pickPointMode = new MapInteractionMode({
+        message: this._getDialogMessage(),
+        buttonText: this._getButtonText(),
+        onCancel: function() {
+            that.terria.mapInteractionModeStack.pop();
+            that._cleanUp();
+        }
+    });
+    this.terria.mapInteractionModeStack.push(pickPointMode);
+    return pickPointMode;
+};
+
+/**
+ * Called after a point has been added, prepares to add and draw another point, as well as updating the dialog.
+ * @private
+ */
+UserDrawing.prototype._prepareToAddNewPoint = function() {
+    var pickPointMode = this._mapInteractionModeUpdate();
+    var that = this;
+
+    knockout.getObservable(pickPointMode, 'pickedFeatures').subscribe(function(pickedFeatures) {
```

Comment:
I guess the idea is it doesn't matter, because the map interaction mode that you're subscribing to is only temporary.